tcms/core/middleware.py

```diff
@@ -5,55 +5,63 @@
     def process_request(self, request):
-        doc_url = "https://kiwitcms.readthedocs.io/en/latest/admin.html#configure-kiwi-s-base-url"
-        site = Site.objects.get(pk=settings.SITE_ID)
+        if resolve(request.path).view_name != "setup":
+            doc_url = "https://kiwitcms.readthedocs.io/en/latest/admin.html#configure-kiwi-s-base-url"
+            try:
+                site = Site.objects.get(pk=settings.SITE_ID)
+            except OperationalError:
+                # Redirect to Setup view
+                return HttpResponseRedirect(reverse("setup"))
 
-        if site.domain == "127.0.0.1:8000":
-            messages.add_message(
-                request,
-                messages.ERROR,
-                mark_safe(  # nosec:B308:B703
-                    _(
-                        "Base URL is not configured! "
-                        'See <a href="%(doc_url)s">documentation</a> and '
-                        '<a href="%(admin_url)s">change it</a>'
-                    )
-                    % {
-                        "doc_url": doc_url,
-                        "admin_url": reverse("admin:sites_site_change", args=[site.pk]),
-                    }
-                ),
-            )
+            if site.domain == "127.0.0.1:8000":
+                messages.add_message(
+                    request,
+                    messages.ERROR,
+                    mark_safe(  # nosec:B308:B703
+                        _(
+                            "Base URL is not configured! "
+                            'See <a href="%(doc_url)s">documentation</a> and '
+                            '<a href="%(admin_url)s">change it</a>'
+                        )
+                        % {
+                            "doc_url": doc_url,
+                            "admin_url": reverse("admin:sites_site_change", args=[site.pk]),
+                        }
+                    ),
+                )
 
 
 class CheckUnappliedMigrationsMiddleware(MiddlewareMixin):
     def process_request(self, request):
-        doc_url = (
-            "https://kiwitcms.readthedocs.io/en/latest/"
-            "installing_docker.html#initial-configuration-of-running-container"
-        )
-        executor = MigrationExecutor(connections[DEFAULT_DB_ALIAS])
-        plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
-        if plan:
-            messages.add_message(
-                request,
-                messages.ERROR,
-                mark_safe(  # nosec:B308:B703
-                    _(
-                        "You have %(unapplied_migration_count)s unapplied migration(s). "
-                        'See <a href="%(doc_url)s">documentation</a>'
-                    )
-                    % {
-                        "unapplied_migration_count": len(plan),
-                        "doc_url": doc_url,
-                    }
-                ),
+        if resolve(request.path).view_name != "setup":
```

Comment:
Don't need the resolve() function. Can match by path.

Reply:
reverse the comparison and return instead of modifying the entire function body.